### PR TITLE
enable race detection for test-infra (except for mungegithub...)

### PR DESCRIPTION
**Pull request by BenTheElder**
In response to discussion at https://github.com/kubernetes/test-infra/issues/4497#issuecomment-328644583, this enables race detection for test-infra except for mungegithub.

### CODE REVIEW

**Comment by BenTheElder** [comment]
we need an area/build or area/bazel or something :thinking:

**Comment by BenTheElder** [comment]
@ixdy https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/4509/pull-test-infra-bazel/6873/ looks suspicious to me:
> ERROR: /workspace/k8s.io/test-infra/experiment/commenter/BUILD:13:1: no such package '@distroless-base//image': Pull command failed: java.io.IOException: Cannot run program "/root/.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/puller/file/puller.par" (in directory

**Comment by BenTheElder** [comment]
@sxptr @rrmh looks like the last test run ran into a race in `deck` ?
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/4509/pull-test-infra-bazel/6875/#prowcmddeckgo_default_test

**Comment by spxtr** [comment]
#4497

File: mungegithub/BUILD [hunk 1]
```diff
@@ -8,12 +8,14 @@ load(
...
 go_binary(
     name = "mungegithub",
+    features = ["-race"],
     library = ":go_default_library",
 )
 
 go_library(
     name = "go_default_library",
     srcs = ["mungegithub.go"],
+    features = ["-race"],
```
Comment (spxtr) [inline mungegithub/BUILD]: You need to set this on the test itself that is failing.

Comment (BenTheElder) [reply mungegithub/BUILD]: Yeah, I think you can set it on the package level but I just did that, I'll remove the extra ones.

File: jobs/config.json [hunk 2]
```diff
@@ -10290,7 +10290,6 @@
       "--build=//...",
       "--install=gubernator/test_requirements.txt",
       "--test=//... //verify:verify-all",
```
Comment (krzyzacy) [inline jobs/config.json]: remove the comma :-) (it's the world of json

File: .bazelrc [hunk 3]
```diff
@@ -1,2 +1,3 @@
 build --workspace_status_command=./print-workspace-status.sh
 run --workspace_status_command=./print-workspace-status.sh
+test --features=race --test_output=errors
```
Comment (krzyzacy) [inline .bazelrc]: nit: new line

File: jobs/config.json (hunk 2 above)
Comment (BenTheElder) [reply jobs/config.json]: done

File: .bazelrc (hunk 3 above)
Comment (BenTheElder) [reply .bazelrc]: done

**Pull request merged**